hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workTracking/WorkTracker.java

```diff
@@ -0,0 +1,67 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workTracking;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.KeeperException;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.cache.zookeeper.NodeCache;
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException;
+import pl.allegro.tech.hermes.domain.subscription.SubscriptionRepository;
+
+import java.util.concurrent.ExecutorService;
+
+import static org.apache.zookeeper.CreateMode.EPHEMERAL;
+
+public class WorkTracker extends NodeCache<SubscriptionAssignmentAware, SubscriptionAssignmentRegistry> {
+    private final SubscriptionRepository subscriptionRepository;
+    private final String supervisorId;
+    private final SubscriptionAssignmentPathSerializer pathSerializer;
+
+
+    public WorkTracker(CuratorFramework curatorClient,
+                       ObjectMapper objectMapper,
+                       String path,
+                       String supervisorId,
+                       ExecutorService executorService,
+                       SubscriptionRepository subscriptionRepository) {
+        super(curatorClient, objectMapper, path, executorService);
+        this.subscriptionRepository = subscriptionRepository;
+        this.supervisorId = supervisorId;
+        this.pathSerializer = new SubscriptionAssignmentPathSerializer(path, supervisorId);
+    }
+
+    public void forceAssignment(Subscription subscription) {
+        askCuratorPolitely(() ->
+                curatorClient.create().creatingParentsIfNeeded().withMode(EPHEMERAL).forPath(pathSerializer.serialize(subscription)));
+    }
+
+    public void dropAssignment(Subscription subscription) {
+        askCuratorPolitely(() ->
+                curatorClient.delete().guaranteed().forPath(pathSerializer.serialize(subscription)));
+    }
+
+    private void askCuratorPolitely(CuratorTask task) {
+        try {
+            task.run();
+        } catch (KeeperException.NodeExistsException | KeeperException.NoNodeException ex) {
```

Comment:
That's actually deliberate behaviour to call this method and not check if node actually exists before both node deletion and node creation. I don't want to warn anybody about this because everything is fine.